lib/curl_gssapi.c

```diff
@@ -44,7 +44,43 @@ OM_uint32 Curl_gss_init_sec_context(
     const bool mutual_auth,
     OM_uint32 *ret_flags)
 {
-  OM_uint32 req_flags = GSS_C_REPLAY_FLAG;
+  gss_cred_id_t creds = GSS_C_NO_CREDENTIAL;
+  OM_uint32 major, discard, req_flags = GSS_C_REPLAY_FLAG;
+
+#ifdef HAVE_GSS_CRED_STORE
+#define MAX_CRED_STORE_ELEMENTS 2
+  gss_key_value_element_desc store_elms[MAX_CRED_STORE_ELEMENTS];
+  gss_key_value_set_desc cred_store;
+
+  const char *ccache = data->set.str[STRING_KRB5CCNAME];
+  const char *ckeytab = data->set.str[STRING_KRB5_CLIENT_KTNAME];
+
+  if(ccache != NULL || ckeytab != NULL) {
+    cred_store.count = 0;
+    cred_store.elements = store_elms;
+
+    if(ccache != NULL) {
+      cred_store.elements[cred_store.count].key = "ccache";
+      cred_store.elements[cred_store.count].value = ccache;
+      cred_store.count++;
+    }
+
+    if(ckeytab != NULL) {
+      cred_store.elements[cred_store.count].key = "client_keytab";
+      cred_store.elements[cred_store.count].value = ckeytab;
+      cred_store.count++;
+    }
```

Comment:
It is worth documenting that you are chaining both. People need to know about this behavior.